src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -75,6 +76,11 @@
  */
 public class XpathQueryGenerator {
 
+    /**
+     * This regexp is used to convert new line to newline tag.
+     */
+    private static final Pattern NEWLINE_TO_TAG = Pattern.compile("[\n]+");
```

Comment:
Do we need the `[]`s and can't just do `\n+`?

Reply:
Please see recent change. `[]` is necessary for current.